temporal-sdk/src/main/java/io/temporal/internal/worker/LocalActivitySlotSupplierQueue.java

```diff
@@ -134,4 +133,40 @@ void submitAttempt(SlotReservationData data, boolean isRetry, LocalActivityAttem
       newExecutionsBackpressureSemaphore.release();
     }
   }
+
+  @Override
+  public boolean isShutdown() {
+    return queueThreadService.isShutdown();
+  }
+
+  @Override
+  public boolean isTerminated() {
+    return queueThreadService.isTerminated();
+  }
+
+  @Override
+  public CompletableFuture<Void> shutdown(ShutdownManager shutdownManager, boolean interruptTasks) {
+    running = false;
```

Comment:
Actually I retract this at the top level shutdown is synchronized so it should be fine.

Reply:
It's a volatile so that part of things ought to be fine. But yeah good point I could catch the interrupt and then just drain the queue.

Reply:
I still think it may be simpler to always interrupt

Reply:
Actually you kinda want to keep this since otherwise you need a different way to know that you should _keep_ looping while alive, which ends up being effectively the same thing as storing this bool.